Source/WebKit/UIProcess/_WKTouchEventGenerator.mm

```diff
@@ -279,23 +280,17 @@ - (void)touchDownAtPoints:(CGPoint*)rawLocations touchCount:(NSUInteger)touchCou
 
     _activePointCount = touchCount;
 
-    for (size_t index = 0; index < locations.size(); ++index)
-        _activePoints[index].point = locations[index];
+    for (auto [i, location] : IndexedRange(locations))
+        _activePoints[i].point = location;
 
     RetainPtr<IOHIDEventRef> eventRef = adoptCF([self _createIOHIDEventType:HandEventTouched]);
     [self _sendHIDEvent:eventRef.get() window:window];
 }
 
 - (void)touchDown:(CGPoint)location touchCount:(NSUInteger)touchCount window:(UIWindow *)window
 {
-    touchCount = std::min(touchCount, HIDMaxTouchCount);
-
-    Vector<CGPoint> locations(touchCount);
-
-    for (NSUInteger index = 0; index < touchCount; ++index)
-        locations[index] = location;
-    
-    [self touchDownAtPoints:locations.data() touchCount:touchCount window:window];
+    Vector<CGPoint> locations(std::min(touchCount, HIDMaxTouchCount), location);
```

Comment:
Should we write `Vector<CGPoint, HIDMaxTouchCount>` to avoid heap memory allocation here?